File: snippets/javascript/leads.js
/**
 * Perfex CRM REST API — Leads examples (Node.js / fetch)
 * Module: https://themesic.com/product/rest-api-module-for-perfex-crm-connect-your-perfex-crm-with-third-party-applications/
 *
 * Node 18+ has global fetch & FormData. No dependencies required.
 */

const BASE = "https://yourdomain.com/api";
const TOKEN = "YOUR_API_TOKEN";

async function listLeads() {
  const res = await fetch(`${BASE}/leads`, {
    headers: { authtoken: TOKEN },
  });
  if (!res.ok) throw new Error(`HTTP ${res.status}`);
  return res.json();
}

async function getLead(id = 1) {
  const res = await fetch(`${BASE}/leads/${id}`, {
    headers: { authtoken: TOKEN },
  });
  if (!res.ok) throw new Error(`HTTP ${res.status}`);
  return res.json();
}

async function createLead() {
  // POST/PUT send multipart/form-data
  const form = new FormData();
  form.append("name", "Jane Doe");
  form.append("email", "jane@example.com");
  form.append("source", "1");
  form.append("status", "1");
  form.append("assigned", "1");
  form.append("phonenumber", "+44 210 7298299");
  form.append("company", "Acme LTD");
  form.append("title", "Procurement Manager");

  const res = await fetch(`${BASE}/leads`, {
    method: "POST",
    headers: { authtoken: TOKEN },
    body: form,
  });
  return { status: res.status, data: await res.json() };
}

async function updateLead(id = 1) {
  const form = new FormData();
  form.append("status", "2");

  const res = await fetch(`${BASE}/leads/${id}`, {
    method: "PUT",
    headers: { authtoken: TOKEN },
    body: form,
  });
  return { status: res.status, data: await res.json() };
}

async function deleteLead(id = 1) {
  const res = await fetch(`${BASE}/leads/${id}`, {
    method: "DELETE",
    headers: { authtoken: TOKEN },
  });
  return { status: res.status, data: await res.json() };
}

(async () => {
  console.log(await listLeads());
  console.log(await getLead(1));
  console.log(await createLead());
  console.log(await updateLead(1));
  console.log(await deleteLead(1));
})();

File: snippets/python/leads.py
"""
Perfex CRM REST API — Leads examples (Python / requests)
Module: https://themesic.com/product/rest-api-module-for-perfex-crm-connect-your-perfex-crm-with-third-party-applications/

Install dependency:  pip install requests
"""
import requests

BASE = "https://yourdomain.com/api"
TOKEN = "YOUR_API_TOKEN"
HEADERS = {"authtoken": TOKEN}


def list_leads():
    r = requests.get(f"{BASE}/leads", headers=HEADERS, timeout=30)
    r.raise_for_status()
    return r.json()


def get_lead(lead_id=1):
    r = requests.get(f"{BASE}/leads/{lead_id}", headers=HEADERS, timeout=30)
    r.raise_for_status()
    return r.json()


def create_lead():
    # POST/PUT send multipart/form-data — pass fields via `data=`
    payload = {
        "name": "Jane Doe",
        "email": "jane@example.com",
        "source": 1,
        "status": 1,
        "assigned": 1,
        "phonenumber": "+44 210 7298299",
        "company": "Acme LTD",
        "title": "Procurement Manager",
    }
    r = requests.post(f"{BASE}/leads", headers=HEADERS, data=payload, timeout=30)
    return r.status_code, r.json()


def update_lead(lead_id=1):
    payload = {"status": 2}
    r = requests.put(f"{BASE}/leads/{lead_id}", headers=HEADERS, data=payload, timeout=30)
    return r.status_code, r.json()


def delete_lead(lead_id=1):
    r = requests.delete(f"{BASE}/leads/{lead_id}", headers=HEADERS, timeout=30)
    return r.status_code, r.json()


if __name__ == "__main__":
    print(list_leads())
    print(get_lead(1))
    print(create_lead())
    print(update_lead(1))
    print(delete_lead(1))
